lib/charms/mongodb_libs/v0/mongodb_tls.py

```diff
@@ -78,6 +79,7 @@ def _on_set_tls_private_key(self, event: ActionEvent) -> None:
             event.fail(str(e))
 
     def _request_certificate(self, scope: str, param: Optional[str]):
+        self.charm.set_secret(scope, "cert", None)
```

Comment:
maybe we shouldn't assign None if the failure happened inside the generate_csr function.
in other words, we can move this line after `self.charm.set_secret(scope, "csr", csr.decode("utf-8"))` line

Reply:
good idea, thanks Mykola :)